Extension/src/LanguageServer/settings.ts

```diff
@@ -312,89 +326,93 @@ export class CppSettings extends Settings {
-    public get defaultWindowsSdkVersion(): string | undefined { return changeBlankStringToUndefined(super.Section.get<string>("default.windowsSdkVersion")); }
-    public get defaultCompileCommands(): string | undefined { return changeBlankStringToUndefined(super.Section.get<string>("default.compileCommands")); }
-    public get defaultForcedInclude(): string[] | undefined { return super.getWithUndefinedDefault<string[]>("default.forcedInclude"); }
-    public get defaultIntelliSenseMode(): string | undefined { return super.Section.get<string>("default.intelliSenseMode"); }
-    public get defaultCompilerPath(): string | undefined { return super.Section.get<string | null>("default.compilerPath") ?? undefined; }
-    public set defaultCompilerPath(value: string | undefined) {
+    public get clangFormatStyle(): string | undefined { return changeBlankStringToUndefined(this.getAsStringOrUndefined("clang_format_style")); }
+    public get clangFormatFallbackStyle(): string | undefined { return changeBlankStringToUndefined(this.getAsString("clang_format_fallbackStyle")); }
+    public get clangFormatSortIncludes(): boolean | undefined { return this.getAsBooleanOrUndefined("clang_format_sortIncludes"); }
+    public get experimentalFeatures(): boolean { return this.getAsString("experimentalFeatures").toLowerCase() === "enabled"; }
+    public get suggestSnippets(): boolean { return this.getAsBoolean("suggestSnippets"); }
+    public get intelliSenseEngine(): string { return this.getAsString("intelliSenseEngine"); }
+    public get intelliSenseEngineFallback(): boolean { return this.getAsString("intelliSenseEngineFallback").toLowerCase() === "enabled"; }
+    public get intelliSenseCachePath(): string | undefined { return changeBlankStringToUndefined(this.getAsStringOrUndefined("intelliSenseCachePath")); }
+    public get intelliSenseCacheSize(): number { return this.getAsNumber("intelliSenseCacheSize"); }
+    public get intelliSenseMemoryLimit(): number { return this.getAsNumber("intelliSenseMemoryLimit"); }
+    public get intelliSenseUpdateDelay(): number { return this.getAsNumber("intelliSenseUpdateDelay"); }
+    public get errorSquiggles(): string { return this.getAsString("errorSquiggles"); }
+    public get inactiveRegionOpacity(): number { return this.getAsNumber("inactiveRegionOpacity"); }
+    public get inactiveRegionForegroundColor(): string | undefined { return changeBlankStringToUndefined(this.getAsStringOrUndefined("inactiveRegionForegroundColor")); }
+    public get inactiveRegionBackgroundColor(): string | undefined { return changeBlankStringToUndefined(this.getAsStringOrUndefined("inactiveRegionBackgroundColor")); }
+    public get autocomplete(): string { return this.getAsString("autocomplete"); }
+    public get autocompleteAddParentheses(): boolean { return this.getAsBoolean("autocompleteAddParentheses"); }
+    public get loggingLevel(): string { return this.getAsString("loggingLevel"); }
```

Comment:
loggingLevel is typed to return a string, but I see it return a `null` -- it's possible there could be similar errors.

<img width="959" alt="image" src="https://github.com/user-attachments/assets/bf99ff0d-4be3-4f73-a052-b6646cb64152">

Something seems wrong with the usage of

```
    private getAsString(settingName: string): string;
    private getAsString(settingName: string, allowNull: boolean): string | null;
    private getAsString(settingName: string, allowNull: boolean = false): string | null {
```

the type checking seems like it gets broken. @bobbrow Is this the recommended pattern or what? i.e. how do we get better type checking to catch errors like this when a non-null is expected?

Reply:
@sean-mcmanus I'm not sure what you're seeing. The first overload says that if only one parameter is passed in, the function will return string.  The implementation should be doing that.  Did you debug it and get null?

Reply:
@bobbrow Yes, the 1 parameter version is passed in but it's still returning null -- is there a way to get the type checker to catch that or is that just something we need to be careful about?

Reply:
> @bobbrow Yes, the 1 parameter version is passed in but it's still returning null

What's the repro?

Reply:
@bobbrow Use a number or null in the loggingLevel setting.

Reply:
I check that the type is string now.